### PR TITLE
Support Midonet in Cloud 7

### PR DESCRIPTION
**Pull request by matelakat**
This PR is based on #558 
Adding crowbar support for midonet

### CODE REVIEW

**Comment by matelakat** [comment]
[P1: Changing crowbar to allow use of Midonet (S2)](https://trello.com/c/OPwYM1Ix/6-p1-changing-crowbar-to-allow-use-of-midonet-s2)

File: chef/cookbooks/neutron/recipes/midonet_agents.rb [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+# limitations under the License.
+#
+
+# we only need to act on compute nodes
+if node.roles.any? { |role| /^nova-compute-/ =~ role }
+  # for /etc/nova/rootwrap.d/
+  package "openstack-nova"
+
+  cookbook_file "/etc/nova/rootwrap.d/network.filters" do
+    source "midonet-network.filters"
```
Comment (dirkmueller) [inline chef/cookbooks/neutron/recipes/midonet_agents.rb]: where is that coming from? feels wrong to install that via crowbar. this should come via some upstream packaging..

File: chef/cookbooks/neutron/templates/default/neutron_lbaas.conf.erb [hunk 2]
```diff
@@ -98,10 +97,13 @@ region_name = <%= @keystone_settings['endpoint_region'] %>
       "LOADBALANCERV2:Haproxy:neutron_lbaas.drivers.haproxy.plugin_driver.HaproxyOnHostPluginDriver:default"
   end
 %>
-service_provider=<%= service_provider %>
 <% elsif @use_lbaas -%>
-service_provider=LOADBALANCER:Haproxy:neutron_lbaas.services.loadbalancer.drivers.haproxy.plugin_driver.HaproxyOnHostPluginDriver:default
+  service_provider = "LOADBALANCER:Haproxy:neutron_lbaas.services.loadbalancer.drivers.haproxy.plugin_driver.HaproxyOnHostPluginDriver:default"
+<% end -%>
+<% if @networking_plugin.eql?("midonet") -%>
+  service_provider = "LOADBALANCER:Midonet:midonet.neutron.services.loadbalancer.driver.MidonetLoadbalancerDriver:default"
```
Comment (nicolasbock) [inline chef/cookbooks/neutron/templates/default/neutron_lbaas.conf.erb]: This is incorrect and will produce two lines of service_provider in the conf file. You could rewrite this section as:

~~~
<% if @use_lbaasv2
       service_provider = case @lbaasv2_driver
                          when "f5"
                              "LOADBALANCERV2:F5Networks:neutron_lbaas.drivers.f5.driver_v2.F5LBaaSV2Driver:default"
                          when nil, "", "haproxy"
                              "LOADBALANCERV2:Haproxy:neutron_lbaas.drivers.haproxy.plugin_driver.HaproxyOnHostPluginDriver:default"
                          end
elsif @use_lbaas
    service_provider = "LOADBALANCER:Haproxy:neutron_lbaas.services.loadbalancer.drivers.haproxy.plugin_driver.HaproxyOnHostPluginDriver:default"
end
if @networking_plugin.eql?("midonet")
    service_provider = "LOADBALANCER:Midonet:midonet.neutron.services.loadbalancer.driver.MidonetLoadbalancerDriver:default"
end
-%>
service_provider = <%= service_provider %>
~~~

**Comment by rsalevsky** [comment]
Can you please prefix all commits like it is already done for the commits from @vuntz?

**Comment by nicolasbock** [comment]
Superseeded by https://github.com/crowbar/crowbar-openstack/pull/614

**Comment by nicolasbock** [comment]
@mjura done in https://github.com/crowbar/crowbar-openstack/pull/614

**Comment by nicolasbock** [comment]
Tried to push to this PR. I can't. Closing again.

**Comment by nicolasbock** [comment]
[P1: Changing crowbar to allow use of Midonet (S2)](https://trello.com/c/OPwYM1Ix/6-p1-changing-crowbar-to-allow-use-of-midonet-s2)

**Pull request closed without merging**